registry/handlers/app.go

```diff
@@ -1072,3 +1058,42 @@ func startUploadPurger(ctx context.Context, storageDriver storagedriver.StorageD
 		}
 	}()
 }
+
+// buildRegexValidationOptions returns a slice of registry options containing
+// rules for regex validation of a certain field (using allow/deny sections)
+func buildRegexValidationOptions(sectionName string, defaultAllowAll bool, allow []string, deny []string, allowOptFunc, denyOptFunc func(r *regexp.Regexp) storage.RegistryOption) []storage.RegistryOption {
+	var options []storage.RegistryOption
+	if len(allow) == 0 && len(deny) == 0 {
+		if defaultAllowAll {
+			options = append(options, storage.ManifestURLsAllowRegexp(regexp.MustCompile("^.*$"))) // allow everything by default
+		} else {
+			options = append(options, storage.ManifestURLsAllowRegexp(regexp.MustCompile("^$"))) // allow nothing by default
+		}
+	} else {
+		if len(allow) > 0 {
+			for i, s := range allow {
+				// Validate via compilation.
+				if _, err := regexp.Compile(s); err != nil {
+					panic(fmt.Sprintf("%s.allow: %s", sectionName, err))
+				}
+				// Wrap with non-capturing group.
```

Comment:
Why? I think all you care about is whether or not `.MatchString` returns true, right?

EDIT: ah I see this was here before and you refactored, but perhaps you know if it has become unnecessary.